flink-clients/src/main/java/org/apache/flink/client/cli/Flip6DefaultCLI.java

```diff
@@ -40,6 +40,7 @@ public Flip6DefaultCLI(Configuration configuration) {
 
 	@Override
 	public boolean isActive(CommandLine commandLine) {
+		this.detachedMode = commandLine.hasOption(CliFrontendParser.DETACHED_OPTION.getOpt());
```

Comment:
I wonder if there is no better place to initialize this field, instead of doing it as undocumented side-effect of something that sounds like an unrelated getter?

Reply:
Why not introduce an abstract method `isDetached(CommandLine)` in `CustomCommandLine`?